client/app/bundles/HelloWorld/reducers/name.jsx

```diff
@@ -0,0 +1,14 @@
+import actionTypes from '../constants/helloWorldConstants';
+
+const name = (state = '', action) => {
```

Comment:
Let's use this for the initial state:
```
{ 
  name: ''
}
```

It's really confusing for me with a single value for the reducer.

Also, we can use the spread operator for the reducer

```
return { ...state, name: text }
```
to ensure we don't mutate state.

@robwise, @alexfedoseev please confirm